x-pack/solutions/security/plugins/security_solution/public/siem_migrations/rules/types.ts

```diff
@@ -12,6 +12,8 @@ export interface RuleMigrationStats extends RuleMigrationTaskStats {
   status: SiemMigrationTaskStatus;
   /** The sequential number of the migration */
   number: number;
+  /** The name of the migration */
+  name?: string;
```

Comment:
Is there any issue if we make the `name` required? The migration will always have one right? Also, we should be able to clean the `number`, which will no longer be used.

Reply:
@kqualters-elastic 
`name` should be required everywhere, and it should be defined in the `RuleMigrationTaskStats` (schema yml). 
In this PR this `RuleMigrationStats` type should be:

```ts
export interface RuleMigrationStats extends RuleMigrationTaskStats {
  status: SiemMigrationTaskStatus; // replace the status by the TS native enum type instead of the zod enum
}
```

- `name` is already included in the `extendsRuleMigrationTaskStats`
- `number` is no longer used, we should clean it.